lib/getdef.c

```diff
@@ -540,12 +541,12 @@ static void def_load (void)
 	/*
 	 * Go through all of the lines in the file.
 	 */
-	while (fgets (buf, (int) sizeof (buf), fp) != NULL) {
+	while (fgets (buf, sizeof (buf), fp) != NULL) {
 
 		/*
 		 * Trim trailing whitespace.
 		 */
-		for (i = (int) strlen (buf) - 1; i >= 0; --i) {
+		for (i = (ptrdiff_t) strlen (buf) - 1; i >= 0; --i) {
```

Comment:
This is, in the end, pointer arithmetic.  strlen(buf) subtracts the pointer to the null byte to the pointer to the start of the buffer.  It is tradition that many standard functions use `size_t` where a `ptrdiff_t` would be more appropriate from a logic point of view, because unsigned integers have special properties that made them more appealing back then.

However, since we subtract one more because we want the position of a character rather than the NUL byte, we can get a negative value, so we can't use `size_t`, which is why we used `int` here.  However, `int` is smaller than `size_t`, and it would be more appropriate to use a type of the same width.  We have `ptrdiff_t` and `ssize_t`.  I chose `ptrdiff_t`, because while both would work, `ssize_t` has a connotation that the value `-1` is reserved to an error code, while in this case we really want the `-1` with the meaning of pointer arithmetic.

You can see that it is indeed pointer arithmetic by the fact that we use `i` to access an array:

```diff
                /*
                 * Trim trailing whitespace.
                 */
-               for (i = (int) strlen (buf) - 1; i >= 0; --i) {
+               for (i = (ptrdiff_t) strlen (buf) - 1; i >= 0; --i) {
                        if (!isspace (buf[i])) {
                                break;
                        }
                }
                i++;
                buf[i] = '\0';
```